### PR TITLE
Feature/direct uploads

### PR DESCRIPTION
**Pull request by nbumbarger**
- Adds a direct-upload endpoint which accepts a File object and adds it to an `uploads` directory.
- Modifies the process-images worker to process local files in addition to downloading remote files, and to remove uploaded files when finished.
- Supports the [addition of direct uploads](https://github.com/hotosm/oam-uploader/tree/feature/direct-uploads) to the uploader form.

### CODE REVIEW

File: worker/process-image.js [hunk 1]
```diff
@@ -36,20 +37,44 @@ function _processImage (s3, scene, url, key, cb) {
...
+    stream
     .pipe(fs.createWriteStream(path))
     .on('finish', function () {
       if (downloadStatus < 200 || downloadStatus >= 400) {
         return callback(new Error('Could not download ' + url +
          '; server responded with status code ' + downloadStatus));
       }
+      // Cleanup local files
+      if (url.includes("file://")) {
+        fs.stat(upload, function (err, stats) {
```
Comment (danielfdsilva) [inline worker/process-image.js]: There's no need to check for the file existence.
Just try to remove it and handle the error if it fails.

It's also suggested by the docs: https://nodejs.org/api/fs.html#fs_fs_stat_path_callback

**Comment by nbumbarger** [comment]
@smit1678 This branch is merged with develop and passing its tests. It should be merged before the [changes to the form](https://github.com/hotosm/oam-uploader/pull/52) that enable file uploads.

File: worker/process-image.js [hunk 2]
```diff
@@ -150,8 +150,7 @@ function translateImage (ext, path, tifPath, callback) {
     '-co', 'COMPRESS=DEFLATE',
     '-co', 'PREDICTOR=2',
     '-co', 'SPARSE_OK=yes',
-    '-co', 'BLOCKXSIZE=512',
-    '-co', 'BLOCKYSIZE=512'
+    '-co', 'BLOCKXSIZE=512'
```
Comment (smit1678) [inline worker/process-image.js]: @nbumbarger Don't we want both X and Y values here?

**Comment by smit1678** [comment]
@nbumbarger @danielfdsilva Before we finalize, we need to make an adjustment to how the metadata is organized. 

Related to https://github.com/hotosm/oam-uploader/pull/52#issuecomment-273257964, we need to put the `license` and `tags` elements into the `properties` object. Similar to how we do `tms` and `thumbnail`.

**Comment by mojodna** [comment]
Why does this not upload direct to S3 (alluded to by #61) using pre-signed URLs if it's available?

**Comment by nbumbarger** [comment]
This allows us to keep a similar workflow for direct uploads and URL-based uploads. Since the file is always on S3, file transfers should be instantaneous and cost-free.

**Comment by nbumbarger** [comment]
@smit1678 @mojodna I added the license and tag properties to this PR, as well as the [sister-PR](https://github.com/hotosm/oam-uploader/pull/52) to the upload form.

**Comment by nbumbarger** [comment]
Closing in favor of https://github.com/hotosm/oam-uploader-api/pull/71, which is built off of the tiling worker branch.

**Pull request closed without merging**